### PR TITLE
realsense2_camera for kinetic

### PR DESCRIPTION
**Pull request by doronhi**
Replaces PR #21431.
Add realsense2 wrapper, realsense2_camera to the kinetic distribution.
This PR depends on PR #21430 that adds librealsense2 installation.

Bloom output:
Increasing version of package(s) in repository realsense2_camera to 2.2.6-1:

upstream repository: https://github.com/IntelRealSense/realsense-ros.git
release repository: https://github.com/IntelRealSense/realsense-ros-release.git
distro file: kinetic/distribution.yaml
bloom version: 0.8.0
previous version for package: null

### CODE REVIEW

**Review by tfoote — CHANGES_REQUESTED** [review]
This depends on #21430 but that's not valid for debian packging so this is not going to work either.

**Comment by doronhi** [comment]
> This depends on #21430 but that's not valid for debian packging so this is not going to work either.

This PR depends now on #21522.

**Review by clalancette — APPROVED** [review]
This looks fine to me, so I'll approve.  I'll let @tfoote review and merge it, however.

**Review by sloretz — DISMISSED** [review]
It looks like the release repository is missing branches needed to create debian packages. If I understand correctly, I think it can be fixed by re-running `bloom-release`. 

@tfoote and @nuclearsandwich explained that this can happen when rosdep keys can't be resolved at the time `bloom-release` was run. It looks like they're all ok now.

<details><summary>realsense2-camera keys</summary>

```
Package 'realsense2-camera' has dependencies:
Run Dependencies:
  rosdep key           => xenial key
  image_transport      => ['ros-kinetic-image-transport']
  cv_bridge            => ['ros-kinetic-cv-bridge']
  nav_msgs             => ['ros-kinetic-nav-msgs']
  nodelet              => ['ros-kinetic-nodelet']
  genmsg               => ['ros-kinetic-genmsg']
  roscpp               => ['ros-kinetic-roscpp']
  sensor_msgs          => ['ros-kinetic-sensor-msgs']
  std_msgs             => ['ros-kinetic-std-msgs']
  message_runtime      => ['ros-kinetic-message-runtime']
  tf                   => ['ros-kinetic-tf']
  ddynamic_reconfigure => ['ros-kinetic-ddynamic-reconfigure']
  diagnostic_updater   => ['ros-kinetic-diagnostic-updater']
  librealsense2        => ['ros-kinetic-librealsense2']
Build and Build Tool Dependencies:
  rosdep key           => xenial key
  image_transport      => ['ros-kinetic-image-transport']
  cv_bridge            => ['ros-kinetic-cv-bridge']
  nav_msgs             => ['ros-kinetic-nav-msgs']
  nodelet              => ['ros-kinetic-nodelet']
  genmsg               => ['ros-kinetic-genmsg']
  roscpp               => ['ros-kinetic-roscpp']
  sensor_msgs          => ['ros-kinetic-sensor-msgs']
  std_msgs             => ['ros-kinetic-std-msgs']
  message_runtime      => ['ros-kinetic-message-runtime']
  tf                   => ['ros-kinetic-tf']
  ddynamic_reconfigure => ['ros-kinetic-ddynamic-reconfigure']
  diagnostic_updater   => ['ros-kinetic-diagnostic-updater']
  librealsense2        => ['ros-kinetic-librealsense2']
  catkin               => ['ros-kinetic-catkin']
```
</details>


<details><summary>realsense2-description keys</summary>

```
Package 'realsense2-description' has dependencies:
Build and Build Tool Dependencies:
  rosdep key           => xenial key
  catkin               => ['ros-kinetic-catkin']
```
</details>

**Comment by doronhi** [comment]
> It looks like the release repository is missing branches needed to create debian packages. If I understand correctly, I think it can be fixed by re-running `bloom-release`.
> 
> @tfoote and @nuclearsandwich explained that this can happen when rosdep keys can't be resolved at the time `bloom-release` was run. It looks like they're all ok now.
> 
> realsense2-camera keys
> realsense2-description keys

Can you please elaborate what branches are missing. I can easily run bloom-release again but I'm not sure how will I know if the problem was corrected.

**Comment by sloretz** [comment]
@doronhi

Current tags on release repo
```
release/kinetic/realsense2_camera/2.2.6-1
release/kinetic/realsense2_description/2.2.6-1
upstream/2.2.6
```

New tags after running bloom-release
```
debian/ros-kinetic-realsense2-camera_2.2.6-2_xenial
debian/ros-kinetic-realsense2-description_2.2.6-2_xenial
```

Current branches on release repo
```
  master
  patches/release/kinetic/realsense2_camera
  patches/release/kinetic/realsense2_description
  release/kinetic/realsense2_camera
  release/kinetic/realsense2_description
  upstream
```

New branches after running bloom release
``` 
  debian/kinetic/realsense2_camera
  debian/kinetic/realsense2_description
  debian/kinetic/xenial/realsense2_camera
  debian/kinetic/xenial/realsense2_description
  patches/debian/kinetic/realsense2_camera
  patches/debian/kinetic/realsense2_description
  patches/debian/kinetic/xenial/realsense2_camera
  patches/debian/kinetic/xenial/realsense2_description
```

The missing branches all come from the debian package generator, so look for branches/tags with `debian` in the name.

**Comment by doronhi** [comment]
@sloretz ,
I re-ran bloom release. The tags in https://github.com/IntelRealSense/realsense-ros-release were changed accordingly.

**Comment by doronhi** [comment]
@sloretz , @tfoote , are the corrections ok?

File: kinetic/distribution.yaml
Comment (sloretz) [inline kinetic/distribution.yaml]: It looks like this version should be [`2.2.6-2`](https://github.com/IntelRealSense/realsense-ros-release/blob/88de910e658188de3170b8f96d7844683f61a71e/README.md#realsense2_camera-kinetic---226-2)

**Review by sloretz — DISMISSED** [review]

File: kinetic/distribution.yaml
Comment (doronhi) [reply kinetic/distribution.yaml]: You are right. Done.

**Comment by clalancette** [comment]
@sloretz @tfoote This is waiting on a re-review from both of you; would you mind doing that?  Thanks.

**Comment by doronhi** [comment]
> @sloretz @tfoote This is waiting on a re-review from both of you; would you mind doing that? Thanks.

The current problem is with librealsense, see #21522 . I need to fix it to build without the examples first.
When this is done, it will be possible to carry on with this PR.

**Comment by tfoote** [comment]
@doronhi Does it make sense to keep this open or should we close this and plan to reopen it later building on top of some of the librealsense updates?

**Comment by doronhi** [comment]
> @doronhi Does it make sense to keep this open or should we close this and plan to reopen it later building on top of some of the librealsense updates?

Looking at [rosdistro/kinetic/distribution.yaml](https://github.com/ros/rosdistro/blob/6f9e3cf8083c9b4bb1eb57aa5f1e06c0a3797fa6/kinetic/distribution.yaml#L10781) I am a little confused already. It seems like this PR is somehow already merged, isn't it?
The current version in rosdistro, as I see using `sudo apt-cache show ros-kinetic-realsense2-camera` is 2.2.6 which is the last one I submitted.

**Comment by clalancette** [comment]
> The current version in rosdistro, as I see using `sudo apt-cache show ros-kinetic-realsense2-camera` is 2.2.6 which is the last one I submitted.

OK, that means that that PR was merged, and not this one.  In that case, I'll close this out.  Feel free to reopen if you think this is in error.

**Pull request closed without merging**
